src/alchemlyb/workflows/abfe.py

```diff
@@ -0,0 +1,691 @@
+        self.dHdl_sample_list = None
+
+    def run(self, skiptime=0, uncorr='dhdl', threshold=50,
+            methods=('MABR', 'BAR', 'TI'), overlap='O_MBAR.pdf',
+            breakdown=True, forwrev=10, *args, **kwargs):
+        ''' The method for running the automatic analysis.
+
+        Parameters
+        ----------
+        skiptime : float
+            Discard data prior to this specified time as 'equilibration' data.
+            Units are specified by the corresponding MD Engine. Default: 0.
+        uncorr : str
+            The observable to be used for the autocorrelation analysis; 'dhdl'
+            (obtained as a sum over those energy components that are changing).
+            Specify as `None` will not uncorrelate the data. Default: 'dhdl'.
+        threshold : int
+            Proceed with correlated samples if the number of uncorrelated samples is
+            found to be less than this number. If 0 is given, the time series
+            analysis will not be performed at all. Default: 50.
+        methods : str or list of str
+            A list of the methods to estimate the free energy with. Default:
+            `('MABR', 'BAR', 'TI')`.
+        overlap : str
+            The filename for the plot of overlap matrix. Default: 'O_MBAR.pdf'.
+        breakdown : bool
+            Plot the free energy differences evaluated for each pair of adjacent
+            states for all methods, including the dH/dlambda curve for TI. Default:
+            ``True``.
+        forwrev : int
+            Plot the free energy change as a function of time in both directions,
+            with the specified number of points in the time plot. The number of time
+            points (an integer) must be provided. Specify as ``None`` will not do
+            the convergence analysis. Default: 10.
+
+        Attributes
+        ----------
+        summary : Dataframe
+            The summary of the free energy estimate.
+        convergence : DataFrame
+            The summary of the convergence results. See
+            :func:`~alchemlyb.convergence.forward_backward_convergence` for
+            further explanation.
+        '''
+        self.read()
+
+        if uncorr is not None:
+            self.preprocess(skiptime=skiptime, uncorr=uncorr,
+                               threshold=threshold)
+        if methods is not None:
+            self.estimate(methods)
+            self.generate_result()
+
+        if overlap is not None:
+            ax = self.plot_overlap_matrix(overlap)
+            plt.close(ax.figure)
+
+        if breakdown:
+            ax = self.plot_ti_dhdl()
+            plt.close(ax.figure)
+            fig = self.plot_dF_state()
+            plt.close(fig)
+            fig = self.plot_dF_state(dF_state='dF_state_long.pdf',
+                                    orientation='landscape')
+            plt.close(fig)
+
+        if forwrev is not None:
+            ax = self.check_convergence(forwrev, estimator='MBAR',
```

Comment:
I think in this case, the user could just go down the manual approach.

```
    >>> workflow = ABFE(units='kcal/mol', software='Gromacs', dir=dir,
    >>>                 prefix='dhdl', suffix='xvg', T=298, outdirectory='./')
    >>> workflow.run(skiptime=10, uncorr='dhdl', threshold=50,
    >>>              methods=('mbar', 'bar', 'ti'), overlap='O_MBAR.pdf',
    >>>              breakdown=True)
    >>> workflow.check_convergence(10, estimator='MBAR', dF_t='dF_t.pdf')
```

Reply:
Well, they could. But what if I actually don't want to run MBAR on my data, e.g.
```python
workflow.run(..., methods=["TI"], ...)
```
so it would be confusing if the convergence analysis was done with a method that I explicitly excluded. I am not saying that this is a very likely case but we don't know which additional estimators we will have. The proposed change `estimators[0]` provides at least some robustness and agency to the user at little cost.

Reply:
I have been thinking about this for some time.

With regard to `estimators[0]`, this is a method that gives better robustness. In some sense, it gives the user agency as it allows the user to choose the method for the convergence. However, in my experience, selecting the first method in methods is not common sense and might be error-prone if BAR is the first estimator.

I think a defensive approach would be that by default convergence analysis is not performed, as it can be quite time-consuming, so `run()` will not trigger convergence analysis.

The user has to do `run(forwrev=10)` to manually trigger the convergence analysis. In the doc for `forwrev`, it will is written that
```
            By default, 'MBAR'
            estimator will be used for convergence analysis. If the dataset
            does not contain u_nk, please run
            meth:`~alchemlyb.workflows.ABFE.check_convergence` manually
            with estimator='TI'.
```
If there is no u_nk, `check_convergence` will give error
```
                    msg = f"u_nk is needed for the f{estimator} estimator. " \
                          f"If the dataset only has dHdl, " \
                          f"run ABFE.check_convergence(estimator='TI') to " \
                          f"use a TI estimator."
                    self.logger.error(msg)
                    raise ValueError(msg)
```
which tells the user that they need to run `check_convergence` to give a different estimator.

Reply:
I am ok with making convergence analysis opt-in as you describe. Just add the appropriate documentation. Add a short explanation _why_ (Auto)MBAR is chosen as the default.